lib/src/references/pubspec_urls.dart

```diff
@@ -58,12 +59,24 @@ Future<PubspecUrlsWithIssues> checkPubspecUrls(PackageContext context) async {
   // and it can be verified as a valid repository.
   final verifiedRepository = await context.repository;
   final isVerifiedRepository = verifiedRepository?.repository != null;
-  if (pubspec.homepage != null &&
-      pubspec.repository == null &&
-      isVerifiedRepository) {
-    var r = repository;
-    repository = homepage;
-    homepage = r;
+  if (pubspec.homepage != null && pubspec.repository == null) {
+    // We may switch these values if the repository has been verified, or
+    // if the verification was not enabled, but the URL parsing recognizes
+    // it as a valid repository.
+    var maySwitch = isVerifiedRepository;
```

Comment:
Is switching not the wrong term here? Could we call this something like
```suggestion
    var canUseHomepageAsRepository = isVerifiedRepository;
```